attribution_scopes.md

```diff
@@ -0,0 +1,229 @@
+
+```jsonc
+// source registration 1 for campaign promoting products 1, 2, and 3 at t=0
+{
+  ..., // existing fields
+  "attribution_scope_limit": 4,
+  "attribution_scopes": ["product1", "product2", "product3"],
+  "max_event_states": 3
+}
+```
+
+```jsonc
+// source registration 2 for campaign promoting only product 2 at t=1
+{
+  ..., // existing fields
+  "attribution_scope_limit": 4,
+  "attribution_scopes": ["product2"],
+  "max_event_states": 3
+}
+```
+
+```jsonc
+// source registration 3 for campaign promoting only product 4 at t=2
+{
+  ..., // existing fields
+  "attribution_scope_limit": 4,
+  "attribution_scopes": ["product4"],
+  "max_event_states": 3
+}
+```
+
+The user then converts at a later time on the destination site and makes a purchase related to product 2 and 3:
+
+```jsonc
+// trigger registration 1 for product related to products 2 and 3 at t=3
+{
+  ..., // existing fields
+  "attribution_scopes": ["product2", "product3"]
+}
+```
+
+In this example the API will start by finding any sources that have an `attribution_scopes` that matches at least one of the trigger registration `attribution_scopes`. In this example that would be source registration 1 and 2. The API will then perform attribution between these two sources. In this example source registration 2 will win the attribution process because it was the most recent source registration. The API caller would receive an attribution report attributing the trigger registration to source registration 2.
+
+## Alternatives Considered
+
+One alternative that was considered was instead of using `attribution_scopes` the API would perform attribution matching on a finer granularity than site (e.g. origin or path). This approach would allow the rest of the API logic to continue working as it does currently. However, the `attribution_scopes` proposal gives users additional flexibility in case their URL structures do not match their logical scope structures. Additionally, matching on finer granularity than site may open additional potential attack vectors and would require additional rate limits.
+
+## Privacy Considerations
+
+### Impact on flexible event-level reporting
+
+The use of `attribution_scopes` will have some impact on the amount of utility that can be extracted from the [Flexible Event-Level](https://github.com/WICG/attribution-reporting-api/blob/main/flexible_event_config.md) reporting proposal. In order to preserve the current privacy levels, the information gain limit that is used to verify acceptable flexible event-level configurations will now include the `attribution_scope_limit` and `max_event_states` values as part of the calculation. In other words, the additional information gain from the `attribution_scope_limit` and `max_event_states` that are chosen will be subtracted from the flexible event-level information gain limit and then used to identify whether the flexible event-level configuration is within the current privacy levels.
```

Comment:
It might warrant a bit of an explanation of why this is the case, since it isn't obvious. Additionally - can we include the formula here? Unless I know how the information gain is impacted, it is hard to judge whether the trade-off is good for any one use-case.

Reply:
+1